docs/practices/when_to_use_private_repository.md

```diff
@@ -0,0 +1,37 @@
+# When to use private Github repository
+
+<blockquote style="background: rgba(255, 200, 0, 0.1); border: 5px solid rgba(255, 200, 0, 0.4);">
+
+The intended audience of the information presented here is developers working
+on the implementation of NEAR.
+
+Are you a security researcher? Please report security vulnerabilities to
+[security@near.org](mailto:security@near.org).
+
+</blockquote>
+
+## TL;DR; 
+
+Limit usage of nearcore-private to sensitive issues that need to be kept confidential
+
+
+## Details
+
+By nature, NEAR protocol ecosystem is community driven effort, welcoming everyone to make contribution; we believe in power of community and openness and bring a greater good. 
+In this regard, we encourage contributors to create an issue/PR in a public space, such as [nearcore](https://github.com/near/nearcore).
+
+However, from time to time, if a sensitive issue is discovered (e.g. security vulnerability), it cannot be tracked publicly; we need to limit the blast radius and conceal the issue from malicious actors. 
+For this purpose, we maintain a private fork, [nearcore-private](https://github.com/near/nearcore-private), of public repository, nearcore, without any changes on top, and track such issues. 
+
+Due to criticality of its contents, the private repository has limited accessbility and is closely monitored by Pagoda protocol engineers. 
+Therefore, special attention is needed on deciding when to use the private repository.
+
+Before creating an issue under nearcore-private, ask the following questions yourself:
+* Does it contain information that can be exploited by malicious actors?
+* Does it need special attention from protocol engineers monitoring the private repository?
```

Comment:
```suggestion
```

I am not sure what kind of information this is trying to convey.  Some folks like to monitor the private repo but they are not required to.  We haven't established any SLAs or processes around what someone should do when an issue is created on the private repo.